src/main/java/com/github/martynagil/drugstoremanagement/service/ComplaintService.java

```diff
@@ -0,0 +1,47 @@
+package com.github.martynagil.drugstoremanagement.service;
+
+import com.github.martynagil.drugstoremanagement.controller.ComplaintDto;
+import com.github.martynagil.drugstoremanagement.controller.ComplaintUpdateDto;
+import com.github.martynagil.drugstoremanagement.model.Complaint;
+import com.github.martynagil.drugstoremanagement.repositories.ComplaintRepository;
+import com.github.martynagil.drugstoremanagement.repositories.ProductRepository;
+import com.github.martynagil.drugstoremanagement.repositories.TransactionRepository;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityNotFoundException;
+
+@Service
+public class ComplaintService {
+
+    private ComplaintRepository complaintRepository;
+    private ProductRepository productRepository;
+    private TransactionRepository transactionRepository;
+
+
+    public ComplaintService(ComplaintRepository complaintRepository) {
+        this.complaintRepository = complaintRepository;
+    }
+
+    public void addComplaint(ComplaintDto complaintDto) {
```

Comment:
Ta metoda powinna walidować czy przypadkiem nie ma już zgłoszonej reklamacji (niezakończonej) do konkretnego towaru - dwie otwarte reklamacje jednocześnie nie mają sensu

Reply:
Myślałam, że walidacje potem będę dodawać

Reply:
Bo podobnie mam wszędzie indziej, np w pracowniku itd

Reply:
poprawiłam ale nie wiem czy to będzie dobrze